### PR TITLE
[11.0][MIG] calendar

### PR DESCRIPTION
**Pull request by yuriqp**
Description of the issue/feature this PR addresses:
migration of calendar module

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/calendar/migrations/11.0.1.0/openupgrade_analysis_work.txt
Comment (StefanRijnhart) [inline addons/calendar/migrations/11.0.1.0/openupgrade_analysis_work.txt:4]: The default is user specific. Can you set user_id to create_uid where user_id is null in the pre-migration?

File: addons/calendar/migrations/11.0.1.0/noupdate_changes.xml
Comment (StefanRijnhart) [inline addons/calendar/migrations/11.0.1.0/noupdate_changes.xml:14]: When we reset noupdate data during a migration, we need to consider if it would be better if the customer's customizations are retained. This is especially true for report and email templates. If you do reset the data, it would be a good idea to unlink the related translations, otherwise the translated template will still be the old version.

Personally, I think in the case of the calendar reminder templates it would be good to take the new version.

**Comment by MiquelRForgeFlow** [comment]
@yuriqp, please, if you do more changes, do a `git pull` before doing them, because I resolved the conflicts.

**Review by StefanRijnhart — APPROVED** [review]
Thanks for the updates!

File: addons/calendar/migrations/11.0.1.0/pre-migration.py
Comment (pedrobaeza) [inline addons/calendar/migrations/11.0.1.0/pre-migration.py:18]: I insist this should be an openupgradelib because it's normal to use this same code across several migration scripts.

Comment (StefanRijnhart) [reply addons/calendar/migrations/11.0.1.0/pre-migration.py:18]: Not sure what you mean by 'insist' but I think it's a good suggestion!

Comment (pedrobaeza) [reply addons/calendar/migrations/11.0.1.0/pre-migration.py:18]: hehe, it was for @yuriqp because he has done it in another PR and I have put the same comment.

Comment (yuriqp) [reply addons/calendar/migrations/11.0.1.0/pre-migration.py:18]: As I mentioned in other PR (https://github.com/OCA/OpenUpgrade/pull/1188) @rruebner already created a PR with this function in openupgradelib (OCA/openupgradelib#87). What should I do, wait until it is merged and then create a new PR to use it in this migration?

Comment (pedrobaeza) [reply addons/calendar/migrations/11.0.1.0/pre-migration.py:18]: When merged, we make another PR calling that function instead.

**Comment by pedrobaeza** [comment]
@yuriqp please make a PR using the recently merged `delete_record_translations` method in openupgradelib.

**Comment by yuriqp** [comment]
I created a new PR using the new `delete_record_translations` method (https://github.com/OCA/OpenUpgrade/pull/1247)

**Pull request merged**
